pkgs/development/tools/build-managers/buildbot/default.nix

```diff
@@ -1,82 +1,57 @@
-{ stdenv, buildPythonApplication, fetchurl, twisted, dateutil, jinja2
-, sqlalchemy_migrate_0_7
-, enableDebugClient ? false, pygobject ? null, pyGtkGlade ? null
+{ stdenv,
+  pythonPackages,
+  fetchurl,
+  plugins ? []
 }:
 
-# enableDebugClient enables "buildbot debugclient", a Gtk-based debug control
-# panel. Its mostly for developers.
-
-assert enableDebugClient -> pygobject != null && pyGtkGlade != null;
-
-buildPythonApplication (rec {
-  name = "buildbot-0.8.12";
-  namePrefix = "";
+pythonPackages.buildPythonApplication (rec {
+  name = "${pname}-${version}";
+  pname = "buildbot";
+  version = "0.9.0rc1";
 
   src = fetchurl {
-    url = "mirror://pypi/b/buildbot/${name}.tar.gz";
-    sha256 = "1mn4h04sp6smr3ahqfflys15cpn13q9mfkapcs2jc4ppvxv6kdn6";
+    url = "mirror://pypi/b/${pname}/${name}.tar.gz";
+    sha256 = "09xsagnfxb8p4a5rqjl69l2q09gnxrba705z8a6sv0jc973c92gh";
   };
 
-  patchPhase =
-    # The code insists on /usr/bin/tail, /usr/bin/make, etc.
-    '' echo "patching erroneous absolute path references..."
-       for i in $(find -name \*.py)
-       do
-         sed -i "$i" \
-             -e "s|/usr/bin/python|$(type -P python)|g ; s|/usr/bin/||g"
-       done
-
-      sed -i 's/==/>=/' setup.py
-    '';
+  propagatedBuildInputs = with pythonPackages; [
+    twisted
+    lz4
+    txrequests
+    dateutil
+    jinja2
+    sqlalchemy
+    sqlalchemy_migrate
+    zope_interface
+    future
+    dateutil
+    autobahn
+    plugins
```

Comment:
if `plugins` is a list of plugins, then you need to append the list with `++`
so `[ twisted ... service-identity ] ++ plugins;`